### PR TITLE
Change glTF skinned mesh reference to a callback

### PR DESCRIPTION
**Pull request by bghgary**
See https://forum.babylonjs.com/t/converting-circular-structure-to-json/30641

### CODE REVIEW

File: packages/dev/loaders/src/glTF/glTFFileLoader.ts [hunk 1]
```diff
@@ -309,6 +310,29 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
...
+    public readonly onSkinLoadedObservable = new Observable<{ node: TransformNode; skinnedNode: TransformNode }>();
+
+    private _onSkinLoadedObserver: Nullable<Observer<{ node: TransformNode; skinnedNode: TransformNode }>>;
+
+    /**
+     * Callback raised when the loader creates a mesh after parsing the glTF properties of the mesh.
+     * Note that the callback is called as soon as the mesh object is created, meaning some data may not have been setup yet for this mesh (vertex data, morph targets, material, ...)
+     */
+    public set onSkinLoaded(callback: (node: TransformNode, skinnedNode: TransformNode) => void) {
+        if (this._onSkinLoadedObserver) {
```
Comment (sebavan) [inline packages/dev/loaders/src/glTF/glTFFileLoader.ts:328]: Are there no issues with several skeletons ?

**Review by sebavan — APPROVED** [review]
Just one tiny question

**Review by deltakosh — APPROVED** [review]

File: packages/dev/loaders/src/glTF/glTFFileLoader.ts [hunk 2]
```diff
@@ -309,6 +310,29 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
...
+     */
+    public readonly onSkinLoadedObservable = new Observable<{ node: TransformNode; skinnedNode: TransformNode }>();
+
+    private _onSkinLoadedObserver: Nullable<Observer<{ node: TransformNode; skinnedNode: TransformNode }>>;
+
+    /**
+     * Callback raised when the loader creates a mesh after parsing the glTF properties of the mesh.
+     * Note that the callback is called as soon as the mesh object is created, meaning some data may not have been setup yet for this mesh (vertex data, morph targets, material, ...)
+     */
+    public set onSkinLoaded(callback: (node: TransformNode, skinnedNode: TransformNode) => void) {
```
Comment (RaananW) [inline packages/dev/loaders/src/glTF/glTFFileLoader.ts:327]: why is this needed? (this and the observer)

File: packages/dev/loaders/src/glTF/glTFFileLoader.ts (hunk 1 above)
Comment (bghgary) [reply packages/dev/loaders/src/glTF/glTFFileLoader.ts]: Not sure what you mean, but this code is removed now.

File: packages/dev/loaders/src/glTF/glTFFileLoader.ts (hunk 2 above)
Comment (bghgary) [reply packages/dev/loaders/src/glTF/glTFFileLoader.ts]: Removed
